types/index.d.ts

```diff
@@ -224,15 +224,7 @@ export type DiffOptions = {
   normalize?: string
 } & PushableOptions
 
-export type diffInfo= {
-  __old?: any
-  __new?: any
-} & LocaleMessages | {
-  __old?: any
-  __new?: any
-}
-
-declare function diff (options: DiffOptions): Promise<diffInfo>
+declare function diff (options: DiffOptions): Promise<LocaleMessages>
```

Comment:
I think that the return I/F of `diff` function should be designed the bellow:

```ts
declare function diff (options: DiffOptions): Promise<Record<string, any>>
```

because LocaleMesaages is an I/F for vue-i18n resources.
This type is indeed the same as `Record<string, any>` in the typedef implementation, but it is used incorrectly. The `diff` should not strictly be a type of `LocaleMessages`, but a type returned by `jsonDiff.diff`. `Record<string, any>` is a compromise.

If it's possible, you might use `infer` to capture the return type of `jsonDiff.diff`.

Reply:
Thank you!
That's right. I fixed it.